### PR TITLE
#938 email validation

### PR DESCRIPTION
**Pull request by dmzaytsev**
For #938 

* fixed regexp 
* added unit tests

### CODE REVIEW

**Comment by karato** [comment]
@dmzaytsev Thanks for your pull request, let me find someone who can review it

**Comment by karato** [comment]
@krzyk do this review

File: netbout-web/src/main/java/com/netbout/dynamo/DyAlias.java [hunk 1]
```diff
@@ -71,9 +71,10 @@
     private final transient Item item;
 
     static {
-        final String valid = "[a-z_\\.@\\-A-Z0-9]";
+        //@checkstyle LineLengthCheck (1 line)
+        final String valid = "([a-z0-9_-]+\\.)*[a-z0-9_-]+@[a-z0-9_-]+(\\.[a-z0-9_-]+)*\\.[a-z]{2,6}";
```
Comment (krzyk) [inline netbout-web/src/main/java/com/netbout/dynamo/DyAlias.java]: @dmzaytsev it won't work for uppercase emails

File: netbout-web/src/test/java/com/netbout/dynamo/DyAliasITCase.java [hunk 2]
```diff
@@ -67,4 +67,41 @@ public void makesAlias() throws Exception {
         );
     }
 
+    /**
+     * DyAlias can reject invalid email.
+     * @throws Exception If there is some problem inside
+     */
+    @Test (expected = Alias.InvalidEmailException.class)
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/dynamo/DyAliasITCase.java]: @dmzaytsev space between annotation and params is rather not standard here :-)

File: netbout-web/src/test/java/com/netbout/dynamo/DyAliasITCase.java [hunk 3]
```diff
@@ -67,4 +67,41 @@ public void makesAlias() throws Exception {
...
+     */
+    @Test
+    public void acceptsValidEmail() throws Exception {
+        final Aliases aliases =
+            new DyBase().user(new URN("urn:test:14")).aliases();
+        final String name = "jack";
+        aliases.add(name);
+        final Alias alias = aliases.iterate().iterator().next();
+        final String[] emails = {
+            "test@domain.com", "test@domain.com!test@domain.com",
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/dynamo/DyAliasITCase.java]: @dmzaytsev I would add a test for dots in the first email part, and sub domains in the other

**Comment by krzyk** [comment]
@dmzaytsev please see my comments

File: netbout-web/src/test/java/com/netbout/dynamo/DyAliasITCase.java (hunk 2 above)
Comment (dmzaytsev) [inline netbout-web/src/test/java/com/netbout/dynamo/DyAliasITCase.java]: @krzyk I wonder why Qulice didn't catch that :)

**Comment by dmzaytsev** [comment]
@krzyk done
please take a look

**Comment by krzyk** [comment]
@dmzaytsev thanks

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by karato** [comment]
@elenavolokhova please, check this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by karato** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@krzyk As far as I can see, only one issue has been fixed after your review. Please, correct me, if I'm wrong.

**Comment by krzyk** [comment]
@elenavolokhova no, there were three, but in two cases the fix was in the lines below my comment

**Comment by elenavolokhova** [comment]
@krzyk Thank you for your help!

**Comment by elenavolokhova** [comment]
@karato Looks good!

**Comment by karato** [comment]
> @karato Looks good!

@elenavolokhova thank you, good quality is always good :)

**Comment by karato** [comment]
@krzyk since quality is good, 10 mins paid to @dmzaytsev (our architect), transaction `75140351`

**Comment by karato** [comment]
@krzyk I added 10 mins to @elenavolokhova (for QA review) in transaction 75151175... Thanks for your contribution, **22 mins** was added to your account, payment ID is `75151188`, 8 hours and 41 mins spent total... you're getting extra minutes here (c=7)... +22 added to your rating, at the moment it is: [+3890](http://www.netbout.com/b/35035?open=rating)

**Pull request merged**